File: soieu/greedy/5585_거스름돈.py
result = 0
# 거스름돈 화폐 종류
MONEY_TYPE = [500, 100, 50, 10, 5, 1]

# 내야할 금액
price = int(input())

# 받아야할 금액
remain = 1000 - price

for money in MONEY_TYPE:
    result += remain // money
    remain = remain % money
    
print(result)

File: soieu/greedy/5585_거스름돈.cpp
#include <iostream>
#include <vector>
using namespace std;


int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);

    vector<int> money = {500, 100, 50, 10, 5, 1};

    int result = 0;

    int remain = 0;

    cin >> remain;
    remain = 1000 - remain;
    for ( int i = 0; i<money.size() ; i++)
    {
        result += remain / money[i];
        remain = remain % money[i];
    }
    cout<<result;
    


    return 0;
}